### ISSUE TITLE
v10.4.0 Component test React 18 warnings with mid sized apps

### ISSUE THREAD

**Issue by muratkeremozcan** [cypress-io/cypress#23169]
### Current behavior

In a simple repo, like a fresh CRA, or small a-la-carte components as in [330+ cypress-react-component-test-examples](https://github.com/muratkeremozcan/cypress-react-component-test-examples) v10.4.0 works as intented; there are no React 18 warnings in the console. 

In mid sized or larger apps the React 18 console warnings may still persist.

Here are 2 apps where the issue is observed. Install and run any component test and view the console.

Repo 1: Kent Dodds' Epic React final app, with Cypress e2e and component tests:
```bash
git clone https://github.com/muratkeremozcan/bookshelf.git
cd bookshelf
npm i
npm run cy:open-ct
```

Repo 2: React Hooks in Action book with Cypress e2e and component tests
```
git clone https://github.com/muratkeremozcan/react-hooks-in-action-with-cypress.git
cd react-hooks-in-action-with-cypress
yarn 
yarn cy:open-ct
```

> clear Cypress cache, remove node_modules, clear npm / yarn cache to ensure clean installs 

![image](https://user-images.githubusercontent.com/34237651/183241026-b484d2f8-2324-4a8a-8600-61f74491fa45.png)

### Desired behavior

Regardless of the app size, there are no React 18 console warnings and we can use new React hooks in the tests.

### Test code to reproduce

Linked above.

### Cypress Version

10.4.0

**Comment by AtofStryker** [comment]
possibly fixed by #23182?

**Comment by rockindahizzy** [comment]
Yes, changing the import in `component.js`

```js
// From
import {mount} from 'cypress/react'

// To
import {mount} from 'cypress/react18'
```

Will clear the warnings, but there is currently [an open issue for React 18 support](https://github.com/cypress-io/cypress/issues/23081).

**Comment by lmiller1990** [comment]
It will be fixed in the next release.

**Comment by muratkeremozcan** [comment]
> Yes, changing the import in `cypress/support/component.js`
> 
> ```js
> // From
> import {mount} from 'cypress/react'
> 
> // To
> import {mount} from 'cypress/react18'
> ```
> 
> Will clear the warnings, but there is currently [an open issue for React 18 support](https://github.com/cypress-io/cypress/issues/23081).

This works. I upgraded React to v18 and Cypress to 10.4.0, and added the new React 18 hooks. 

But, headless local or CI execution is so slow that CI times out, and local is dreadful. There has to be something not right.

Please make sure to try this in a large repo.

You can reproduce through this PR https://github.com/muratkeremozcan/cypress-react-component-test-examples/pull/280 .

```bash
git clone https://github.com/muratkeremozcan/cypress-react-component-test-examples.git
git checkout feat/react18-hooks
yarn 
yarn cy:run-ct
# very slow
```

**Comment by lmiller1990** [comment]
Headless execution for Component Testing will improve by about 40% with an experimental flag that is landing soon.

- Pull Request: https://github.com/cypress-io/cypress/pull/23104
- Issue with context: https://github.com/cypress-io/cypress/issues/22353

**Comment by lmiller1990** [comment]
205 specs
Before: 42m
After: 4m

Wow, unreal. I had no idea webpack was *this* slow - I've tested other [webpack repos](https://github.com/cypress-io/cypress/issues/22353#issuecomment-1170807871) which saw improvements around 2x. I am not sure what's going on with your project but this is very weird, I can't believe it takes 40 minutes - other projects we have internally using webpack don't exhibit anything near the slowness you are experiencing.

Final numbers (run on my Macbook arm64)

```
# experimentalSingleTabRunMode: false 
yarn cy:run-ct --config video=false --browser chrome  
17.88s user 9.95s system 11% cpu 4:00.11 total

# experimentalSingleTabRunMode: true 
yarn cy:run-ct --config video=false --browser chrome  
2585.68s user 125.62s system 105% cpu 42:42.53 total
```

**Comment by muratkeremozcan** [comment]
> 

Thanks a ton for giving it a try.

On main, with React 17, the local run time is about 15 mins. In CI it is all right, under 4 mins with parallelization https://dashboard.cypress.io/projects/62pyqm/runs/933/specs. 

If the experimental feature fixes the issue occurence with React 18, and increases the speed 2-3 times, works for me.

**Comment by lmiller1990** [comment]
Great, I hope we can ship the experimental flag with 10.5. If it goes smoothly, we can make this the default in the future. 🤞 Keep your eyes on https://github.com/cypress-io/cypress/pull/23104 - just waiting for a few internal reviews.

**Comment by muratkeremozcan** [comment]
> #23104

I tried `experimentalSingleTabRunMode` , React 18, and the new hooks with v10.6.0. Ran out of parallelization while wrestling the changes, but got one last run https://dashboard.cypress.io/projects/62pyqm/runs/991/specs. The parallel duration of 340+ tests came down from ~5 mins to ~3 mins. 

If it works in this repo, it should work in most production apps.

https://github.com/muratkeremozcan/cypress-react-component-test-examples/pull/297
